src/init.cpp

```diff
@@ -1337,20 +1337,15 @@ bool AppInit2(Config &config, thread_group &threadGroup)
     RegisterNodeSignals(GetNodeSignals());
 
     // sanitize comments per BIP-0014, format user agent and check total size
-    std::vector<string> uacomments;
+    std::vector<string> uacomments = {};
     for (string &cmt : mapMultiArgs["-uacomment"])
     {
         if (cmt != SanitizeString(cmt, SAFE_CHARS_UA_COMMENT))
             return InitError(strprintf(_("User Agent comment (%s) contains unsafe characters."), cmt));
         uacomments.push_back(SanitizeString(cmt, SAFE_CHARS_UA_COMMENT));
     }
-    strSubVersion = FormatSubVersion(CLIENT_NAME, CLIENT_VERSION, uacomments);
-    if (strSubVersion.size() > MAX_SUBVERSION_LENGTH)
-    {
-        return InitError(strprintf(_("Total length of network version string (%i) exceeds maximum length (%i). Reduce "
-                                     "the number or size of uacomments."),
-            strSubVersion.size(), MAX_SUBVERSION_LENGTH));
-    }
+
+    std::string strSubVersion = FormatSubVersion(CLIENT_NAME, CLIENT_VERSION, BUComments);
```

Comment:
In order to still give the user immediate feedback, without requiring them to look at the log file, I think it makes sense to still have a pop-up via `InitWarning` in lieu of the `InitError` that's being replaced.  Suggest something like

```
    std::string strSubVersion = FormatSubVersion(CLIENT_NAME, CLIENT_VERSION, BUComments);
    if (strSubVersion.size() == MAX_SUBVERSION_LENGTH)
        InitWarning(strprintf(_("Total length of network version string with uacomments added exceeded "
            "the maximum length (%i) and have been truncated.  Reduce the number or size of uacomments "
            "to avoid truncation."), MAX_SUBVERSION_LENGTH));
```

NOTE: This solution isn't perfect because the user could have entered in uacomments which exactly hits the max length without the need for truncation.  To be able to determine if truncation was actually performed we'd have to modify FormatSubVersion to be able to return this information in addition to the strSubVersion.

Reply:
ok this is neat. will add even if it's not "prefect" like you mentioned